src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -196,6 +196,11 @@ private bool ParsePreface(ReadableBuffer readableBuffer, out ReadCursor consumed
 
         private Task ProcessFrameAsync<TContext>(IHttpApplication<TContext> application)
         {
+            if (_incomingFrame.StreamId != 0 && (_incomingFrame.StreamId & 1) == 0)
```

Comment:
This one needs a comment. It's ok to quote the spec. Even better if you link to the section.